system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -283,7 +319,7 @@ class TopSitesEdit extends React.Component {
           <FormattedMessage id="edit_topsites_button_text" />
         </button>
       </div>
-      {this.state.showEditModal &&
+      {this.state.showEditModal && !this.state.showAddForm && !this.state.showEditForm &&
```

Comment:
nit: maybe refactor this in a single constant at the top, seems excessively verbose to have it here.